src/xpk/core/tests/unit/test_nodepool.py

```diff
@@ -0,0 +1,90 @@
+from src.xpk.core.nodepool import get_desired_node_pool_names
+
+CLUSTER_NAME = "running-cucumber"
+
+
+def node_pool_name(number: int) -> str:
+    return f'{CLUSTER_NAME}-np-{number}'
+
+
+def test_compute_desired_node_pool_names_with_desired_larger_than_existing():
+    result = get_desired_node_pool_names(
+        existing_node_pool_names=[
+            node_pool_name(0)
+        ],
+        cluster_name=CLUSTER_NAME,
+        desired_node_pool_count=2
+    )
+
+    expected_result = [
+        node_pool_name(0),
+        node_pool_name(1)
+    ]
+    assert set(result) == set(expected_result)
+
+
+def test_compute_desired_node_pool_names_with_desired_smaller_than_existing():
+    result = get_desired_node_pool_names(
+        existing_node_pool_names=[
+            node_pool_name(0),
+            node_pool_name(1)
+        ],
+        cluster_name=CLUSTER_NAME,
+        desired_node_pool_count=1
+    )
+
+    expected_result = [node_pool_name(0)]
+    assert set(result) == set(expected_result)
+
+
+def test_compute_desired_node_pool_names_with_consecutive_numbers_missing():
+    result = get_desired_node_pool_names(
+        existing_node_pool_names=[
+            node_pool_name(0),
+            node_pool_name(3)
+        ],
+        cluster_name=CLUSTER_NAME,
+        desired_node_pool_count=3
+    )
+
+    expected_result = [
+        node_pool_name(0),
+        node_pool_name(1),
+        node_pool_name(3)
+    ]
+    assert set(result) == set(expected_result)
+
+
+def test_compute_desired_node_pool_names_with_consecutive_numbers_missing_and_desired_equal_to_existing():
+    result = get_desired_node_pool_names(
+        existing_node_pool_names=[
+            node_pool_name(0),
+            node_pool_name(3)
+        ],
+        cluster_name=CLUSTER_NAME,
+        desired_node_pool_count=2
+    )
+
+    expected_result = [
+        node_pool_name(0),
+        node_pool_name(3)
+    ]
+    assert set(result) == set(expected_result)
+
+
+def test_compute_desired_node_pool_names_with_unknown_node_pools():
+    result = get_desired_node_pool_names(
+        existing_node_pool_names=[
+            node_pool_name(0),
+            node_pool_name(3),
+            "unknown-node-pool"
```

Comment:
nit: could you put this non-standard name as first in the list. It doesn't matter for the current code, but I feel it might have caught some potential bugs in the future.